### DISCUSSION TITLE
How not to redirect any `/api/` calls?

### DISCUSSION THREAD

**Discussion by binarykitchen** [remix-run/react-router#11882] [General]
My app does plenty of HTTP API and Web Socket calls to the backend.

React-router is causing too many invalid redirects.

All those API calls start with the `/api/` path. How can I configure react-router not to redirect any of these back to index.html?

This is the webpack-dev-server configuration I'm using:
```ts
    historyApiFallback: {
      disableDotRule: true,
      // I'd like to see any request rewrites
      verbose: true,
    },
```

Thanks to the `verbose` flag, I see this in the console:
```
Rewriting GET /video/123/preview/type/mp4/1723455007509 to /index.html
```

Read the official documentation about four times, still no answer about that. I do not want to use a loader as these API calls can happen in between route changes.

Thanks for any hints

**Comment by ksachin7** [comment]
Add rewrites(used to exclude certain paths from being rewritten) see this: [historyApiFallback](https://webpack.js.org/configuration/dev-server/#devserverhistoryapifallback)
